indico/modules/events/sessions/operations.py

```diff
@@ -81,8 +82,20 @@ def delete_session(event_session):
 
 def update_session_block(session_block, data):
     """Update a session block with data passed in the `data` argument"""
+    start_dt = data.pop('start_dt', None)
+    if start_dt is not None:
+        update_timetable_entry(session_block.timetable_entry, {'start_dt': start_dt})
     session_block.populate_from_dict(data)
     db.session.flush()
     session_block.event_new.log(EventLogRealm.management, EventLogKind.change, 'Sessions',
                                 'Session block "{}" has been updated'.format(session_block.title), session.user)
     logger.info('Session block %s modified by %s', session_block, session.user)
+
+
+def delete_session_block(session_block):
+    session_ = session_block.session
+    for contribution in session_block.contributions:
+        contribution.session_block = None
```

Comment:
you also need to delete the timetable entry of each contribution (or you could handle that in `delete_timetable_entry` if the entry is for a session block)